x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -176,7 +180,9 @@ protected final boolean assertCurrentThreadMayLoadSnapshot() {
      *
      * @return true if the snapshot was loaded by executing this method, false otherwise
      */
-    public boolean loadSnapshot() {
+    public boolean loadSnapshot(RecoveryState recoveryState) {
+        assert recoveryState != null;
+        assert recoveryState instanceof SearchableSnapshotRecoveryState;
```

Comment:
This assumes that noone is touching the index setting `index.recovery.type`. Perhaps we should also throw an explicit exception here when assertions are not enabled?